src/marked.js

```diff
@@ -76,34 +70,30 @@ function marked(src, opt, callback) {
 
     delete opt.highlight;
 
-    if (!pending) return done();
+    if (!tokens.length) return done();
 
-    for (; i < tokens.length; i++) {
-      (function(token) {
-        if (token.type !== 'code') {
-          return --pending || done();
-        }
-        return highlight(token.text, token.lang, function(err, code) {
-          if (err) return done(err);
```

Comment:
I realized the callback sent to marked could be called multiple times because of this line. I'm not sure if that was intentional or a bug but I left it in case anyone relies on it being called for multiple highlighting errors.